### PR TITLE
feat(autonomy): deterministic lane-stop gate + operator STOP notification

### PR DESCRIPTION
**Pull request by kyle-sexton**
## What

Ships the two cheap, additive hooks the operator ratified as the leave-week safety set for the
"operational resilience" umbrella (#535): **member 3** (deterministic lane-stop gate) and **member 4**
(operator notification on a genuine lane stop). Both land inside the `autonomy` plugin — its first
`hooks/` directory — and touch nothing else.

## Member 3 — deterministic lane-stop gate (`hooks/lane-stop-gate.sh`, `Stop` event)

"A lane that stops itself before its goal is met is a bug" was only a prompt admonition. This hook
fires on every stop attempt of an **opted-in** lane and structurally intercepts it:

- **Completion is signaled deterministically** (a shell hook cannot re-run the `/goal` evaluator
  model) — either the exact **sentinel token** (default `LANE-STOP-OK`, matched standalone-only) in
  the agent's final message, or the existence of a configured **marker file**. This is the
  settings-scoped, cross-session sibling of `/goal`'s session-only completion condition (#481); its
  own docs name "a prompt-based Stop hook" as exactly this alternative.
- **Not signaled → the first stop is blocked** with a re-injected completion self-check
  (`decision:"block"` + reason). The reason directly counters the fabricated-context-percentage
  premature-stop failure (#576/#577): a self-estimated "~50% context", a turn count, or a vague sense
  of "enough" is explicitly stated to not be completion.
- **Still stops after the one nudge → allowed** (never wedged) and the operator is notified (member 4).

Safety properties: **default OFF** (a Stop-blocking hook must never engage for an interactive
session); **fail-open** on unreadable stdin, missing `jq`, or a non-`Stop` event (a `SubagentStop`
never trips it); bounded against runaway by the `stop_hook_active` one-block guard plus Claude Code's
own consecutive-block cap. **Scope:** catches a graceful **self-stop** only — a closed laptop, a
killed process, or `/loop` expiry emit no `Stop` event and are members 1/2's job.

## Member 4 — operator notification (`hooks/lane-notify.sh`)

Fired by the gate at the genuine-stop juncture. OS-native toast (macOS/Linux) + best-effort terminal
bell + OSC 9, **local machine only** (no remote/Slack/push primitive exists in the marketplace yet).
It **reimplements** rather than sources `desktop-notification` because a `Stop` hook's stdout is parsed
for `decision`/`reason` and cannot use the `terminalSequence` field that plugin's `Notification` hook
relies on (a genuinely different emission path: direct `/dev/tty`), and because cache-isolated plugins
cannot source each other at runtime. No separate repeated-failure counter was built — a lane that
keeps stopping simply re-fires this notification, and `StopFailure` API-error telemetry is already
owned by `claude-ops` (avoids that plugin's in-flight PRs entirely).

## Files

- `plugins/autonomy/hooks/{lane-stop-gate.sh, lane-notify.sh, hooks.json, hook-utils.sh}` (new)
- `plugins/autonomy/hooks/{lane-stop-gate.test.sh, lane-notify.test.sh}` (new, 25 black-box cases)
- `plugins/autonomy/.claude-plugin/plugin.json` (0.9.0 → 0.10.0, six `userConfig` options)
- `plugins/autonomy/{CHANGELOG.md, README.md}`

## Verification (local)

shellcheck (0.11.0) + shfmt (v3.13.1) + markdownlint clean; both test suites pass (15 + 10);
`sync-hook-utils --check` (13 copies), `check-silent-skips`, `check-cross-plugin-source-drift`,
`check-changelog-parity` (`--check` + `--check-bump`), and `validate-plugins` all pass.

## For the reviewer — please double-check

- **False-positive risk (gate blocks a legitimate stop):** a lane that genuinely finished but forgot
  to emit the sentinel / touch the marker gets exactly one blocked turn, then is allowed to stop with
  a notification. Bounded and non-fatal, and only for opted-in lanes. Acceptable?
- **False-negative risk (gate allows a premature stop):** the agent emits the sentinel while
  hallucinating completion. Mitigated by requiring a *deliberate* standalone token (much higher bar
  than a passive stop) and by the marker-file alternative; residual risk documented.
- **`stop_hook_active` reset semantics** decide whether the gate re-arms for another nudge on a later
  natural stop (per-attempt reset) or nudges only once per session (latched). The code and safety
  bound are correct under either reading; the docs are worded to not over-claim a per-cluster reset.
  If you know the authoritative semantics, confirm the wording.
- **`Stop` vs `SubagentStop` firing:** the hook early-allows unless `hook_event_name == "Stop"`, so a
  Task-tool worker finishing can never trip the lane gate regardless of how the platform routes the
  registration.

## Issue linkage

No linked issue — this PR partially addresses the umbrella #535 (members 3+4 only) and closes nothing.
The umbrella stays open with members 1/2/5 deferred; see Related.

## Related

Refs #535 (members 3+4 per the ratified 2026-07-20T02:17:15Z triage-routing decision — members 1/2/5
remain deferred, not in scope).

### CODE REVIEW

**Comment by kyle-sexton** [comment]
**Orchestrator review pass (rule 10 full-diff-read + post-green review).**

Reviewed the full diff personally (Stop-hook gate = high blast radius). Logic review: fail-open paths correct (empty stdin, missing jq, non-Stop event all allow-and-exit), default-OFF verified, one-nudge-then-notify bounded correctly against `stop_hook_active`, sentinel standalone-token matching is sound, argv-safe `osascript`/`notify-send` invocation, control-byte stripping on notification text, test coverage for exactly the fail-open/fail-closed edge cases that matter (12 cases, all the right ones). No DEFER-FORBIDDEN-class finding — logic is safe.

**One real finding, fix requested before flipping ready:**

README.md's example lane-launcher invocation is not valid Claude Code CLI syntax:

```
claude --config lane_stop_gate_enabled=true --config lane_stop_gate_marker=.lane-complete ...
```

Verified against https://code.claude.com/docs/en/plugins-reference.md: `--config <key=value>` is documented as an option of the **`claude plugin install <plugin>`** subcommand ("Set a userConfig option declared in the plugin's manifest... Repeat the flag to set multiple options") — it sets the value persistently at install/enable time, written to `pluginConfigs[<plugin-id>].options` in settings.json. It is not documented as a flag on the top-level `claude` session-launch invocation shown in the README's example.

The correct per-session-override mechanism per the same doc page is `claude --settings '<inline JSON>'`, which can include a `pluginConfigs` override, e.g. (verify the exact `<plugin-id>` key format empirically or against docs before finalizing — the docs show a `formatter-my-marketplace`-style slug for a marketplace-scoped plugin in a different context (`CLAUDE_PLUGIN_DATA`), which may or may not be the same key `pluginConfigs` uses):

```
claude --settings '{"pluginConfigs":{"autonomy":{"options":{"lane_stop_gate_enabled":true,"lane_stop_gate_marker":".lane-complete"}}}}'
```

Please fix the README example to a verified-working invocation (test it empirically if the docs are ambiguous on the exact `pluginConfigs` key) — this is documentation a real lane launcher will copy verbatim, so it needs to actually work.

**Comment by kyle-sexton** [comment]
Full CI green (24/24 checks) after both fix rounds. Flipping to ready.

Summary for the record: rule-10 full-diff-read found the core logic (Stop-hook gate, notification, tests) correct with no DEFER-FORBIDDEN-class issue. Post-green review surfaced one doc-accuracy defect (invalid `--config` CLI example); round 1 fixed it correctly but introduced a portability-contract violation (hardcoded this org's marketplace name into an org-agnostic plugin); round 2 fixed that too, verified via the actual plugin-gate validator locally before pushing. This closes out the review pass for members 3+4 of #535 — members 1/2/5 remain open/deferred on the umbrella issue, per the ratified scope.

File: plugins/autonomy/hooks/lane-stop-gate.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+# Signal 1 — the sentinel token in the agent's final message. Matched only as a
+# standalone token (not a substring of a longer alnum run), so a message that
+# merely discusses the mechanism does not incidentally authorize the stop.
+SENTINEL="${CLAUDE_PLUGIN_OPTION_LANE_STOP_GATE_SENTINEL:-LANE-STOP-OK}"
+if [[ -n "$SENTINEL" ]]; then
+  LAST=$(printf '%s' "$INPUT" | jq -r '.last_assistant_message // ""' 2>/dev/null)
+  # Escape any regex metacharacters in the (configurable) sentinel before use.
+  SENTINEL_RE=$(printf '%s' "$SENTINEL" | sed 's/[][\.^$*+?(){}|/]/\\&/g')
+  if printf '%s' "$LAST" | grep -qE "(^|[^A-Za-z0-9-])${SENTINEL_RE}([^A-Za-z0-9-]|$)"; then
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/autonomy/hooks/lane-stop-gate.sh]: | # | Finding | Classification | Evidence | Reacted |
|---|---------|----------------|----------|---------|
| 1 | Require a line-level completion token — inline mention/negation of the sentinel should not fire the gate | **VALID — fixed in `a6335f4a`** | Detection at `lane-stop-gate.sh:71` matched the sentinel as a standalone *word* anywhere in the final message, so `"I should not emit LANE-STOP-OK yet"` set `SIGNALED=1`. Changed the match to require the token **alone on its own line** (`^[[:space:]]*<sentinel>[[:space:]]*$`), which is exactly the `"emit the exact token … on its own line"` instruction the block reason already gives — detection and the documented contract now agree. New regression tests: Case 13 (`"I should not emit LANE-STOP-OK yet"` → **blocked**) and Case 14 (token alone on an indented line amid other lines → **allowed**). Full suite 17/17 green. | 👍 |

Scope note (so this reply does not over-claim): the line-level fix closes the *inline mention/negation* false-positive you flagged. It does **not** address the separate concern that the block reason discloses the literal token, letting an agent emit it on its own line with no further work — that is the security-review IMPORTANT finding, a distinct design tradeoff (hiding the token from the nudge would degrade a legitimately-complete lane's ability to discover the completion protocol), classified separately and left for the author to decide. The gate remains a bounded one-nudge-then-allow+notify speed-bump by design, not a hard stop.

**Comment by kyle-sexton** [comment]
**Classification of the HEAD (`73035c6`) review findings** — PR-Review ([comment](https://github.com/melodic-software/claude-code-plugins/pull/969#issuecomment-5043762383)) and Security-Review ([comment](https://github.com/melodic-software/claude-code-plugins/pull/969#issuecomment-5043763390)). The two 08:21 passes referenced the stale commit `687229b4` and are superseded by these HEAD passes; the earlier PR-Review "hooks registration wiring" High was not repeated at HEAD (resolved between passes).

| # | Finding | Severity | Classification | Evidence |
|---|---------|----------|----------------|----------|
| 1 | Line-level completion token — inline mention/negation of the sentinel bypasses the gate (codex P2) | P2 | **VALID — fixed in `a6335f4a`** | Detection now requires the token alone on its own line; regression tests added (Case 13 blocked / Case 14 allowed). See the [threaded reply on the codex comment](https://github.com/melodic-software/claude-code-plugins/pull/969#discussion_r3628870319). |
| 2 | Sentinel disclosed in the block reason ("repeat the token we gave you") | IMPORTANT | **VALID (defer) — author design call** | Real but bounded: the gate is one-nudge-then-allow+notify by design, so a determined agent's stop is allowed after one nudge *regardless* of this — the residual impact of disclosure is mainly that the agent skips the second attempt and thus the operator alert, not a new hard-gate bypass. Hiding the literal token from the nudge trades away a legitimately-complete lane's ability to discover the completion protocol (it may not be able to read `userConfig`). Whether to make that trade is a design decision for the author, not a clear-cut defect; not fixed here to avoid silently choosing a priority. |
| 3 | Unvalidated CWD (`..`) in relative marker-path construction (L83–84) | MEDIUM / SUGGESTION | **VALID (defer) — defense-in-depth** | `cwd` originates from Claude Code's own hook payload; impact is a false-negative (premature stop), not privilege escalation, and only on the relative-marker branch. Legitimate hardening via the existing `hook::physical_path`, out of scope for this line-level correctness fix. |
| 4 | Empty-sentinel behavior not documented in `lane_stop_gate_sentinel` `userConfig` | Low | **VALID (defer) — doc nit** | One-sentence parallel to the `lane_stop_gate_marker` note ("Empty string disables the token-based signal"). No behavior change; cheap follow-up. |
| 5 | `git -C "$CWD"` without `--`/path validation (L105) | Low / SUGGESTION | **VALID (defer) — graceful today** | A `-`-leading `cwd` makes `git -C` fail and `BRANCH` falls back to `"unknown"`; cheap explicit-intent hardening, no current defect. |

This pass fixed the one core-purpose correctness finding (#1 — the gate firing on a mention of its own token). Findings #2–#5 are a genuine design tradeoff (#2) plus defense-in-depth/doc follow-ups (#3–#5) with no `DEFER-FORBIDDEN` security class among them (all are false-negative-only, or design/doc), so they are classified and left for the author rather than swept into this line-level fix.

File: plugins/autonomy/hooks/lane-stop-gate.sh [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  exit 0
+fi
+
+# First stop attempt without a completion signal → block once and re-inject the
+# completion self-check. This directly counters the fabricated-context-percentage
+# premature-stop failure (#576/#577): a self-estimated "~50% context" is not a
+# completion condition. Emitted as the documented Stop stdout decision.
+REASON="Autonomy lane-stop gate: you attempted to stop, but this lane's completion condition is not yet signaled. A lane that stops itself before its stated goal is met is a bug. Do NOT stop on a self-estimated context percentage, a turn count, or a vague sense that enough was done — none of those is completion. Either (1) continue working toward the lane's stated goal, or (2) if the goal is genuinely and verifiably met, declare completion by emitting the exact token ${SENTINEL} on its own line (or by creating the configured completion-marker file), then stop. This is your one automated nudge; if you stop again without signaling completion, the operator will be alerted that the lane went down."
+
+jq -nc --arg r "$REASON" '{decision:"block", reason:$r}'
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/autonomy/hooks/lane-stop-gate.sh:164]: | # | Finding | Classification | Evidence | Reacted |
|---|---------|---------------|----------|---------|
| 1 | Gate never calls `hook::emit_telemetry`, so lane-stop outcomes are invisible to `HOOK_TELEMETRY_SINK` | **VALID (defer)** — real convention gap; implementation is an author design call, not a mechanical consistency fix | Telemetry emission IS an established, consistent convention: 26 non-test call sites across 12+ plugins (formatters, guardrails, claude-ops audits, desktop-notification), and `hook::emit_telemetry` is already present in `plugins/autonomy/hooks/hook-utils.sh:391`. But no autonomy hook emits yet, and every one of the 26 emitters ships a published `docs/conventions/hook-telemetry/data/<hook>.schema.json` — there is none for this gate. Adding it here is therefore not a one-line trailer; it requires new design. | 👍 |

**Why deferred to the author rather than fixed in this round** (same posture as this PR's other design-tradeoff finding). Wiring this correctly means making, not guessing, these decisions:

1. **New published data schema.** The convention (`docs/conventions/hook-telemetry/README.md`) makes `data/<hook>.schema.json` the source of truth, discovered by the envelope's `hook` value. A `lane-stop-gate` producer needs its own schema artifact authored and committed.
2. **Outcome→status mapping has a gap.** The documented status set is `ok | error | skipped | blocked`. `blocked` covers the first-attempt block (L123) cleanly, but the outcome this finding most wants to measure — the nudged, still-stopping down-lane that is allowed with an operator alert (L105–114) — has no home in that set. Distinguishing it from an ordinary completion-signaled allow (L95, `ok`) requires a designed `data.outcome` field, which is a schema decision.
3. **Fail-open exit-path restructure.** The gate has multiple early `exit 0` paths (default-off L42, fail-open stdin/non-Stop L46/L58, silent allow L95, notify+allow L114, block L123) and captures no `EPOCHREALTIME` start. Emitting per-outcome needs either an EXIT trap over an outcome variable or emit calls at each branch, plus a `START` snapshot — a deliberate restructure of a fail-open, security-sensitive hook, not a drop-in.
4. **Security-conscious field selection.** This exact file carries three open security findings (sentinel disclosure, CWD/marker path traversal, `git -C` CWD). The `data` payload must be designed to NOT surface the sentinel token or an unsanitized CWD into the sink envelope.

Net: the convention gap is real and worth closing, but it is a scoped design task (schema + status vocabulary + exit-path/telemetry restructure + security review of the payload), so it is left for the author to design rather than have a review-round worker invent a schema.

**Comment by kyle-sexton** [comment]
🔒 merge-drive lane claiming this PR for on-branch fix work this cycle (amendment-round: 16, worker tier). Will fix clear branch-owned CI/bot findings, defer non-DEFER-FORBIDDEN items with reply+resolve, push, then re-check the merge gate.

**Comment by kyle-sexton** [comment]
**Classification of the HEAD (`a6335f4a`) review findings** — PR-Review ([comment](https://github.com/melodic-software/claude-code-plugins/pull/969#issuecomment-5044253811)) and Security-Review ([comment](https://github.com/melodic-software/claude-code-plugins/pull/969#issuecomment-5044254089)). These two passes ran against `a6335f4a` itself and were not covered by the earlier `73035c6`-era classification; this closes that gap.

| # | Finding | Severity | Classification | Evidence |
|---|---------|----------|----------------|----------|
| 1 | `plugin.json` `lane_stop_gate_sentinel` description stale after `a6335f4a` — still says "Match is standalone-token only" (raised independently by both passes) | ⚠️ new defect / SUGGESTION | **VALID — fixed in `e64b9882`** | Verified at `plugins/autonomy/.claude-plugin/plugin.json:34` on the PR head: the description described the pre-`a6335f4a` word-boundary behavior while detection at `lane-stop-gate.sh:74` requires the token alone on its own line. Wording now mirrors the language already on the branch — `README.md:117` ("alone on its own line") and `CHANGELOG.md:22` ("matched only when alone on its own line") — so the one surface a configuring operator reads agrees with the gate's own block reason. Doc-only; no behavior change. |
| 2 | Sentinel disclosed in the block reason | IMPORTANT | **Unchanged — VALID (defer), author design call** | Both passes converge on the prior classification: the PR-Review explicitly records "the accepted design trade-off the PR description documents … **No new action required**". Not re-decided here. |
| 3 | Unvalidated CWD (`..`) in relative marker-path construction (L83–88) | SUGGESTION | **Unchanged — VALID (defer), defense-in-depth** | Reported "still unaddressed", not escalated; impact remains false-negative only (premature stop allowed), not privilege escalation. |
| 4 | `git -C "$CWD"` without absolute-path guard (L108) | Low / SUGGESTION | **Unchanged — VALID (defer)** | Reported "still unaddressed", graceful today (`BRANCH` falls back to `unknown`). |
| 5 | Empty-sentinel behavior undocumented in `userConfig` | Low | **Unchanged — VALID (defer), doc nit** | Both passes suggest bundling it into the #1 description edit. Deliberately left out: it was already classified defer by the author and is a pre-existing gap, not a defect this branch introduced — the branch-owned regression is what was fixed. It can ride the same field in the author's follow-up. |

Everything else in the security pass landed under "Items reviewed and found acceptable" (OSC 9 / AppleScript / `notify-send` injection surfaces, `jq --arg` encoding, sentinel ERE escaping, hooks registration, default-OFF, SubagentStop guard) — no action.

**Not merge-eligible this cycle regardless:** the branch is behind `main` and GitHub reports `BLOCKED`; a base refresh is an orchestrator-only action, so it is reported rather than performed here.

File: plugins/autonomy/hooks/lane-stop-gate.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/autonomy/hooks/lane-stop-gate.sh:164]: **Implemented in `ac576670`** — the deferral above is superseded; the four design decisions it enumerated have now been made deliberately rather than guessed:

1. **Published data schema**: `docs/conventions/hook-telemetry/data/lane-stop-gate.schema.json`, discovered from the envelope `hook: "lane-stop-gate"`; implementers table row added to the convention README.
2. **Outcome→status mapping**: the documented status set is kept as-is; the down-lane allow the finding most wants to measure is distinguished by a designed `data.outcome` field — `blocked`/`nudged` (first structural nudge), `ok`/`completion-signaled` with `data.signal: sentinel|marker` (legitimate stop), `ok`/`stopped-after-nudge` (allowed with the operator alert) — so the local alert is correlatable in the sink.
3. **Exit paths**: only the three **evaluated** outcomes emit; the default-off and fail-open exits stay silent (they are pre-evaluation, and emitting on every interactive default-off stop would be noise). `EPOCHREALTIME` start is snapshotted after the opt-in gate, so `duration_ms` covers real gate work and interactive sessions pay nothing.
4. **Security-conscious payload**: `data` is a closed fixed vocabulary — it never carries the sentinel token value, the marker path, the cwd, or the branch, so the envelope cannot leak the completion token or lane-identifying paths into the sink (asserted by a test that greps the captured envelope for the token).

Emission is the convention's fire-and-forget `hook::emit_telemetry` (no-op without `HOOK_TELEMETRY_SINK`; internal stdout/stderr fully redirected, so the Stop `decision`/`reason` stdout contract is untouched). New test cases capture the envelope via a stub sink for all three outcomes. Suite 24/24 green; CHANGELOG and plugin README updated.

File: plugins/autonomy/hooks/lane-stop-gate.sh [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+# is exactly the "emit the exact token ... on its own line" instruction the block
+# reason gives. Requiring a dedicated line — not merely a standalone word — means
+# a message that only mentions or negates the token inline (e.g. "I should not
+# emit LANE-STOP-OK yet") does not authorize the stop.
+SENTINEL="${CLAUDE_PLUGIN_OPTION_LANE_STOP_GATE_SENTINEL:-LANE-STOP-OK}"
+if [[ -n "$SENTINEL" ]]; then
+  LAST=$(printf '%s' "$INPUT" | jq -r '.last_assistant_message // ""' 2>/dev/null)
+  # Escape any regex metacharacters in the (configurable) sentinel before use.
+  SENTINEL_RE=$(printf '%s' "$SENTINEL" | sed 's/[][\.^$*+?(){}|/]/\\&/g')
+  if printf '%s' "$LAST" | grep -qE "^[[:space:]]*${SENTINEL_RE}[[:space:]]*$"; then
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/autonomy/hooks/lane-stop-gate.sh]: **VALID — fixed in `ac576670`.** Reproduced exactly as described: with `pipefail`, `printf '%s' "$LAST" | grep -q` returns rc 141 (SIGPIPE) when the sentinel matches early in a message longer than the pipe buffer — verified empirically (old pattern rc=141, fix rc=0 with the sentinel followed by ~120KB of text). A completed lane would have been blocked, then reported as down on its next stop.

Fix: the match now uses a here-string (`grep -qE "..." <<<"$LAST"`) — no pipeline, so an early `grep -q` exit can never discard a match. New regression coverage: a ~120KB final message with the sentinel on the first line must be **allowed**, plus a control case (same long tail, no sentinel) that must still **block** — proving the long payload genuinely reaches the gate rather than passing via a fail-open path. Full suite 24/24 green.

File: plugins/autonomy/hooks/lane-stop-gate.sh [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+set -uo pipefail
+
+# shellcheck source=hook-utils.sh
+source "$(dirname "${BASH_SOURCE[0]}")/hook-utils.sh"
+# shellcheck source=lane-notify.sh
+source "$(dirname "${BASH_SOURCE[0]}")/lane-notify.sh"
+
+# Default-OFF opt-in (NOT hook::check_enabled, which defaults ON when unset).
+[[ "${CLAUDE_PLUGIN_OPTION_LANE_STOP_GATE_ENABLED:-false}" == "true" ]] || exit 0
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/autonomy/hooks/lane-stop-gate.sh:43]: Addressed in 83b75b07: the claude-ops lane launcher now takes an optional per-lane `settings` JSON object in the lanes config and passes it verbatim as `claude --settings` on every start/restart (session-only, verified against the current CLI reference; validated as an object at preflight so a malformed value skips the lane). The gate opt-in via the `pluginConfigs` override is the documented motivating example in `context/config.md`, so standing lanes get the gate and the stop notification from tracked lane config rather than persistent global configuration. claude-ops bumped to 0.20.0.

File: plugins/autonomy/hooks/lane-stop-gate.sh [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  /* | [A-Za-z]:[/\\]*) ;;
+  *)
+    CWD=$(printf '%s' "$INPUT" | jq -r '.cwd // ""' 2>/dev/null | tr -d '\r')
+    [[ -n "$CWD" ]] && MARKER="${CWD%/}/$MARKER"
+    ;;
+  esac
+  if [[ -f "$MARKER" ]]; then
+    SIGNALED=1
+    SIGNAL="marker"
+  fi
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/autonomy/hooks/lane-stop-gate.sh:128]: Addressed in 83b75b07: the marker is now consumed (deleted) when it authorizes a stop — one marker, one authorized stop — so a file left by a prior completed run in the same checkout never authorizes stops of a later run. Test coverage asserts both the consumption and that the consumed path no longer authorizes the next stop.
